### PR TITLE
New element Sponsor + Rules + Tests

### PR DESCRIPTION
**Pull request by everton-rosario**
This PR

* [x] Creates a new Element Sponsor
* [x] Creates test cases for the new Element Sponsor 
* [x] Creates transformation rule for the new element Sponsor 
* [x] Creates test cases and content for the new rule

### CODE REVIEW

File: src/Facebook/InstantArticles/Elements/ListElement.php [hunk 1]
```diff
@@ -97,6 +97,7 @@ public function addItem($new_item)
     public function withItems($new_items)
     {
         Type::enforceArrayOf($new_items, [ListItem::getClassName(), Type::STRING]);
+        $this->items = [];
```
Comment (diegoquinteiro) [inline src/Facebook/InstantArticles/Elements/ListElement.php]: nice catch! ;)

File: src/Facebook/InstantArticles/Elements/Sponsor.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+ *
+ * This source code is licensed under the license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+namespace Facebook\InstantArticles\Elements;
+
+use Facebook\InstantArticles\Validators\Type;
+
+/**
+ * Class BrandedContentList that represents a simple HTML list
```
Comment (diegoquinteiro) [inline src/Facebook/InstantArticles/Elements/Sponsor.php]: Did you mean `Class Sponsor`?

File: src/Facebook/InstantArticles/Elements/Sponsor.php [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+    /**
+     * Factory method for a Sponsor.
+     *
+     * @return Sponsor the new instance.
+     */
+    public static function create()
+    {
+        $list = new self();
+        $list->disableOrdered();
```
Comment (diegoquinteiro) [inline src/Facebook/InstantArticles/Elements/Sponsor.php]: Use `parent::createUnordered()`

File: src/Facebook/InstantArticles/Elements/Sponsor.php [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+        return $list;
+    }
+
+    /**
+     * Sets the page url sponsor. Overrides the previous set URL.
```
Comment (diegoquinteiro) [inline src/Facebook/InstantArticles/Elements/Sponsor.php]: Sets the page url *of the* sponsor

File: src/Facebook/InstantArticles/Elements/Sponsor.php [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+ */
+class Sponsor extends ListElement
+{
+    /**
+     * @var string page URL.
+     */
+    private $page_url;
+
+    private function __construct()
+    {
```
Comment (diegoquinteiro) [inline src/Facebook/InstantArticles/Elements/Sponsor.php]: Change parent constructor to protected and remove the constructor.

File: src/Facebook/InstantArticles/Elements/Sponsor.php [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+/**
+ * Class BrandedContentList that represents a simple HTML list
+ *
+ * Example:
+ * <ul class="op-sponsors">
+ *     <li><a href="http://facebook.com/your-sponsor" rel="facebook"></a></li>
+ * </ul>
+ *
+ */
+class Sponsor extends ListElement
```
Comment (diegoquinteiro) [inline src/Facebook/InstantArticles/Elements/Sponsor.php]: We don't need to extend ListElement here.

**Comment by chrisackermann** [comment]
lgtm!

**Pull request merged**